statsmodels/multivariate/manova.py

```diff
@@ -0,0 +1,176 @@
+    hypotheses testing and M is the transformation matrix for transforming the
+    dependent variables in y.
+
+    Algorithm:
+        H = (L * u * M)' * (L * u * M)
+        E = M' * sscpr * M
+
+    .. [1] ftp://public.dhe.ibm.com/software/analytics/spss/documentation/statistics/20.0/en/client/Manuals/IBM_SPSS_Statistics_Algorithms.pdf
+
+    Parameters
+    ----------
+    """ + _hypotheses_doc +
+    """
+    fittedvalues : tuple
+        Output of ``fit_manova``
+    exog_names : a list of string
+    endog_names : a list of string
+
+    Returns
+    -------
+    a dict of manova tables
+    """)
+
+
+class MANOVA(Model):
+    """
+    Multivariate analysis of variance
+
+
+    Parameters
+    ----------
+    endog : array-like
+        Dependent variables (DV). A n_sample x n_y_var array where n_sample is
+        the number of observations and n_y_var is the number of DV.
+
+    exog : array-like
+        Independent variables (IV). A n_sample x n_x_var array where n is the
+        number of observations and n_x_var is the number of IV. An intercept is
+        not included by default and should be added by the user (models
+        specified using a formula include an intercept by default)
+
+    .. [1] ftp://public.dhe.ibm.com/software/analytics/spss/documentation/statistics/20.0/en/client/Manuals/IBM_SPSS_Statistics_Algorithms.pdf
+
+    Attributes
+    -----------
+    df_resid : float
+        The number of observation `n` minus the number of IV `q`.
+    sscpr : array
+        Sums of squares and cross-products of residuals
+    endog : array
+        See Parameters.
+    exog : array
+        See Parameters.
+    design_info : patsy.DesignInfo
+        Contain design info for the independent variables if model is
+        constructed using `from_formula`
+
+    """
+    def __init__(self, endog, exog, missing='none', hasconst=None, **kwargs):
+        super(MANOVA, self).__init__(endog, exog, **kwargs)
+        self.fittedmod = _manova_fit(self.exog, self.endog)
+
+    def mv_test(self, hypotheses=None):
+        if hypotheses is None:
+            if (hasattr(self, 'data') and self.data is not None and
+                        self.data.design_info is not None):
+                terms = self.data.design_info.term_name_slices
+                hypotheses = []
+                for key in terms:
+                    L_contrast = np.eye(self.exog.shape[1])[terms[key], :]
+                    hypotheses.append([key, L_contrast, None])
+            else:
+                hypotheses = []
+                for i in range(self.exog.shape[1]):
+                    name = 'x%d' % (i)
+                    L = np.zeros([1, self.exog.shape[1]])
+                    L[i] = 1
+                    hypotheses.append([name, L, None])
+
+        results = _manova_test(hypotheses, self.fittedmod,self.exog_names,
+                               self.endog_names)
+
+        return MultivariateTestResults(results)
```

Comment:
general question: Are the hypotheses L and M arrays attached to the MultivariateTestResults instance?
I think being able to get those would be useful. Those could then also be used for unit tests, e.g. comparing from_formula with no formula MANOVA results.